backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -695,6 +704,53 @@ public FutureExperimentRunDAO(
                                                     hyperparams.get(builder.getId()))),
                                     executor);
 
+                            final var futureHyperparamsFromConfigBlobs =
+                                InternalFuture.completedInternalFuture(
+                                        populateConnectionsBasedOnPrivileges)
+                                    .thenCompose(
+                                        populateConnectionsBasedOnPrivileges -> {
+                                          if (populateConnectionsBasedOnPrivileges) {
+                                            return getAllowedEntitiesByResourceType(
+                                                ModelDBActionEnum.ModelDBServiceActions.READ,
+                                                ModelDBResourceEnum.ModelDBServiceResourceTypes
+                                                    .REPOSITORY);
+                                          } else {
+                                            return InternalFuture.completedInternalFuture(
+                                                new ArrayList<>());
+                                          }
+                                        },
+                                        executor)
+                                    .thenCompose(
+                                        selfAllowedRepositoryIds -> {
+                                          if (selfAllowedRepositoryIds == null
+                                              || selfAllowedRepositoryIds.isEmpty()) {
+                                            return InternalFuture.completedInternalFuture(
+                                                    new ArrayList<
+                                                        AbstractMap.SimpleEntry<
+                                                            String, KeyValue>>())
+                                                .thenApply(MapSubtypes::from, executor);
```

Comment:
Why do we need this? You already know that the array will be empty because you just created it. Reading this code makes me confused

Reply:
Simplify logic and reduce one line